remote_attestation/rust/src/handshaker.rs

```diff
@@ -448,7 +454,6 @@ impl ServerHandshaker {
                 .as_ref()
                 .context("Couldn't get TEE certificate")?;
 
-            let additional_info = self.additional_info.clone();
             let attestation_info =
                 create_attestation_info(signer, additional_info.as_ref(), tee_certificate)
```

Comment:
As a side note, would it work to just take a reference here instead of cloning it? It seems we are cloning and getting a ref immediately anyways, I don't remember if there is a reason for that or it's just an accident.

cc @ipetr0v

Reply:
Yes we ref the clone, but the actual clone is passed to `ServerIdentity` a few lines down.

Reply:
To be fair we can clone it there, (which means we no longer need to clone for every step only this one). I added a commit for that. It's an improvement but we still need the other changes (stop keeping a clone in the Handshaker struct)